lib/main.dart

```diff
@@ -14,16 +17,20 @@ void main() async {
 }
 
 class ECellApp extends StatelessWidget {
+  final SharedPreferences sharedPreferences = sl.get<SharedPreferences>();
   @override
   Widget build(BuildContext context) {
     SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp]);
+    String token = sharedPreferences.getString(S.tokenKeySharedPreferences);
     return MaterialApp(
       title: "ECellApp",
       theme: AppTheme.themeData(context),
-      home: BlocProvider(
-        create: (context) => LoginCubit(APILoginRepository()),
-        child: LoginScreen(),
-      ),
+      home: token != null
+          ? HomeScreen()
+          : BlocProvider(
+              create: (context) => LoginCubit(APILoginRepository()),
+              child: LoginScreen(),
+            ),
```

Comment:
This is supposed to be done in splash screen. First make that, then do this. For now, leave it as it was.

Reply:
removed this for now